a10_octavia/api/drivers/driver.py

```diff
@@ -291,6 +291,12 @@ def validate_flavor(self, flavor_dict):
                         raise Exception('end-address is required for nat-pool-list flavor')
                     if 'netmask' not in nat:
                         raise Exception('netmask is required for nat-pool-list flavor')
+            if 'deployment' in flavor_dict:
+                deployment = flavor_dict['deployment']
+                if ('dsr_type' not in deployment or
```

Comment:
We may have other attributes in deployment flavor in the feature. So, for flexibility maybe we can't assume 'dsr_type' is necessary for 'deployment'.

Reply:
fixed